app/models/bot/smooch.rb

```diff
@@ -941,7 +941,7 @@ def self.get_id_from_send_response(response)
     (RequestStore.store[:smooch_bot_provider] == 'TURN' || RequestStore.store[:smooch_bot_provider] == 'CAPI') ? response_body&.dig('messages', 0, 'id') : response&.body&.message&.id
   end
 
-  def self.save_smooch_response(response, pm, query_date, fallback_template = nil, lang = 'en', custom = {}, expire = nil)
+  def self.save_smooch_response(response, pm, query_date, fallback_template = nil, lang = nil, custom = {}, expire = nil)
```

Comment:
question: could you help me understand why you moved the "en" default out of this method signature? it seems like now we have more places to track down where it's being set, and I'm not seeing the benefit

Reply:
We just have two types of cases today:

(1) Newsletters delivered: For that, we want to pass in the languages of the newsletters.

(2) Messages that were sent but failed because of the 24-hours window problem and for which we want to send a template message... WhatsApp requires a language to be present, so for those cases it's better to be sure it's `en` otherwise a template message is not sent.

But since we're relying on that Smooch payload to also save `TiplineMessage`'s, I prefer to fallback to the user language than to English, which should be a default only for the template messages.  Currently, it won't make a difference, but I'm thinking about future cases.

Since you worked on this more than me recently, I am happy to revert this last commit if you think it's more clear.

Reply:
I revert this last commit for now.

Reply:
gotcha! yeah, I think it's good to think about future cases, but would be clearer to make that change when/if we need it. thanks for taking this comment into consideration!